dspace-api/src/main/java/org/dspace/external/provider/impl/SHERPAv2PublisherDataProvider.java

```diff
@@ -0,0 +1,361 @@
+            method.setConfig(RequestConfig.custom()
+                .setConnectionRequestTimeout(timeout)
+                .setConnectTimeout(timeout)
+                .setSocketTimeout(timeout)
+                .build());
+
+            // Execute the method
+            HttpResponse response = client.execute(method);
+            int statusCode = response.getStatusLine().getStatusCode();
+
+            if (statusCode != HttpStatus.SC_OK) {
+                sherpaResponse = new SHERPAPublisherResponse("SHERPA/RoMEO return not OK status: "
+                    + statusCode);
+            }
+
+            HttpEntity responseBody = response.getEntity();
+
+            // Get InputStream from API response and parse JSON
+            if (null != responseBody) {
+                InputStream content = null;
+                try {
+                    content = responseBody.getContent();
+                    sherpaResponse = new SHERPAPublisherResponse(content, SHERPAPublisherResponse.SHERPAFormat.JSON);
+                } catch (IOException e) {
+                    log.error("Encountered exception while contacting SHERPA/RoMEO: " + e.getMessage(), e);
+                } finally {
+                    if (content != null) {
+                        content.close();
+                    }
+                }
+            } else {
+                sherpaResponse = new SHERPAPublisherResponse("SHERPA/RoMEO returned no response");
+            }
+        } catch (Exception e) {
+            log.error("SHERPA/RoMEO query failed: ", e);
+        }
+
+        if (sherpaResponse == null) {
+            sherpaResponse = new SHERPAPublisherResponse("Error processing the SHERPA/RoMEO answer");
+        }
+        if (CollectionUtils.isNotEmpty(sherpaResponse.getPublishers())) {
+            SHERPAPublisher sherpaPublisher = sherpaResponse.getPublishers().get(0);
+            // Construct external data object from returned publisher
+            ExternalDataObject externalDataObject = constructExternalDataObjectFromSherpaPublisher(sherpaPublisher);
+            return Optional.of(externalDataObject);
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Search SHERPA v2 API for publisher results based on a 'contains word' query for publisher name
+     * @param query The query for the search
+     * @param start The start of the search
+     * @param limit The max amount of records to be returned by the search
+     * @return a list of external data objects
+     */
+    @Override
+    public List<ExternalDataObject> searchExternalDataObjects(String query, int start, int limit) {
+        // query args to add to SHERPA/RoMEO request URL
+        HttpGet get = null;
+        try {
+            get = constructHttpGet(query, start, limit);
+            HttpClient hc = new DefaultHttpClient();
+            HttpResponse response = hc.execute(get);
+            if (response.getStatusLine().getStatusCode() == 200) {
```

Comment:
Can we use "HttpStatus.SC_OK" instead of 200 ?

Reply:
Done! See commit# 839159b